### ISSUE TITLE
Install fails on Windows 11

### ISSUE THREAD

**Issue by brandstaetter** [cloudflightio/dockerinwsl#53]
Installation fails on Windows 11, complains about wsl2 not being installed, despite it being the default version.

In cmd.exe:

```
Microsoft Windows [Version 10.0.22621.819]
(c) Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\micro>winget search dockerinwsl
Name        Id                      Version        Source
---------------------------------------------------------------
DockerInWSL Cloudflight.DockerInWSL 1.3.119.220810 cloudflight

C:\Users\micro>winget install Cloudflight.DockerInWSL
Found DockerInWSL [Cloudflight.DockerInWSL] Version 1.3.119.220810
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Installer failed with exit code: 1603
Installer log is available at: C:\Users\micro\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Cloudflight.DockerInWSL.1.3.119.220810-2022-11-29-05-42-30.512.log
```

System info:

```
Edition			Windows 11 Home
Version			22H2
Installiert am		04.‎11.‎2022
Betriebssystembuild	22621.819
Leistung		Windows Feature Experience Pack 1000.22636.1000.0
```

Error message:

```
---------------------------
DockerInWSL Installation 
---------------------------
WSL2 not installed!

Check your windows-updates and run 'wsl --install' in an elevated shell, reboot and start this install again.
---------------------------
OK   
---------------------------
```

Screenshot of error message:

![image](https://user-images.githubusercontent.com/644820/204440917-871da8d1-8b08-45fa-b22e-f2c2354e5312.png)

Log file:

[WinGet-Cloudflight.DockerInWSL.1.3.119.220810-2022-11-29-05-42-30.512.log](https://github.com/cloudflightio/dockerinwsl/files/10109801/WinGet-Cloudflight.DockerInWSL.1.3.119.220810-2022-11-29-05-42-30.512.log)

**Comment by mriedmann** [comment]
Very likely a bug in the check script https://github.com/cloudflightio/dockerinwsl/blob/master/msi/install/check.ps1 
Currently I have no Win11 env available => a PR would be very much appreciated.

**Comment by brandstaetter** [comment]
It worked after I manually downloaded and ran https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi -- before that there was no `kernel` file at this location.

**Comment by mriedmann** [comment]
Ok so maybe we should just improve the error message?

**Comment by brandstaetter** [comment]
I would like to gather more input first, to either change/improve the detection or to identify something missing in the documentation.

I think on Win11 the kernel might be in a different location, because it def. worked before I installed the kernel manually.

I'd like to try it on a fresh Win11  install again, will have to check how best to do that...

**Comment by gipfelen** [comment]
When I tried to install wsl_update_x64.msi I got the message, that WSL is not installed, even though the ubuntu WSL was already working. After a restart the error message disapeared and i was able to install dockerinwsl on Windows 11.

**Comment by mriedmann** [comment]
ok this obviously sounds like a bug in the actual check ... I think this would be fixed if someone could get this script https://github.com/cloudflightio/dockerinwsl/blob/master/msi/install/check.ps1 to work with win11 correctly. I have no win11 right now. Maybe someone is willing to help?

**Comment by brandstaetter** [comment]
Ok so on my current machine I find the `kernel` file here: `C:\Program Files\WindowsApps\MicrosoftCorporationII.WindowsSubsystemForLinux_1.1.6.0_x64__8wekyb3d8bbwe\tools\kernel` but I guess it does not help much...
On the other hand, after a failed install I have: 
![image](https://user-images.githubusercontent.com/644820/229152564-92134abf-7226-4ff5-b753-88580b1bed76.png)
So maybe we should just detect if it's windows 11 and be fine with that?

**Comment by Pystronic** [comment]
I tried to setup the project today on Win 11, and i have the same problem. My wsl distributions and kernel position match with your description.

**Comment by mandoway** [comment]
Hi, I just had the issue that WSL could not connect to the internet causing image pulls to fail.
This is caused by Windows generating a wrong DNS server config for WSL and can be fixed by following this SO post: https://stackoverflow.com/questions/62314789/no-internet-connection-on-wsl-ubuntu-windows-subsystem-for-linux

TLDR: 
Edit `/etc/resolv.conf` and change the nameserver to a proper one.
e.g. 8.8.8.8 (google) or 1.1.1.1 (cloudflare)

**Comment by DevNoteHQ** [comment]
> @DevNoteHQ , see the above comment from @brandstaetter. That actually worked for me on the P16. I also enabled HyperV from windows features, and WSL.

Yeah i read that already. Installed the kernel update and then it worked on the P16. But I didn't have to do that on my private Notebook. Maybe I should check whether the kernel is installed at the mentioned location on my private Notebook. Would be strange though since I did the same steps on both Notebooks...

**Comment by horneac** [comment]
I had the same issue with "please install wsl..." 
I couldn't run the wsl_update_x64.msi as it said the same thing, even after countless reboots and uninstall/install of the wsl and dockerinwsl. 
I found for some reason that in "turn windows features on or off" that I had "Virtual Machine platform" checked and "Windows subsystem for linux" unchecked, even thow wsl was working and I could access ubuntu. 
I did the following:
* Go to "turn windows features on or off"
* disable both "Virtual machine platform" and "Windows subsystem for linux"
* reboot
* enable both of them
* reboot again
* now installing docker in wsl works without the error
Maybe just enabling the "Windows subsystem for linux" in the first place could have also solved the problem

**Comment by DevNoteHQ** [comment]
> I had the same issue with "please install wsl..."  I couldn't run the wsl_update_x64.msi as it said the same thing, even after countless reboots and uninstall/install of the wsl and dockerinwsl.  I found for some reason that in "turn windows features on or off" that I had "Virtual Machine platform" checked and "Windows subsystem for linux" unchecked, even thow wsl was working and I could access ubuntu.  I did the following:
> 
> * Go to "turn windows features on or off"
> * disable both "Virtual machine platform" and "Windows subsystem for linux"
> * reboot
> * enable both of them
> * reboot again
> * now installing docker in wsl works without the error
>   Maybe just enabling the "Windows subsystem for linux" in the first place could have also solved the problem

"Windows Subsystem for Linux" is for WSL 1 and is not needed for WSL 2, which is why your WSL was working even without it. The docker-installer could have a check on that built in though...
